olive/model/hf_utils.py

```diff
@@ -25,6 +27,137 @@ class HFComponent(ConfigBase):
+    # other kwargs
+    extra_args: Dict = None
+
+    @validator("torch_dtype", pre=True)
+    def validate_torch_dtype(cls, v):
+        if isinstance(v, torch.dtype):
+            v = str(v).replace("torch.", "")
+        return v
+
+    @validator("device_map", pre=True)
+    def validate_device_map(cls, v):
+        if isinstance(v, torch.device):
+            v = cls.device_to_str(v)
+        elif isinstance(v, dict):
+            v = {k: cls.device_to_str(v) for k, v in v.items()}
+        return v
+
+    @validator("quantization_config", pre=True, always=True)
+    def validate_quantization_config(cls, v, values):
+        if "quantization_method" not in values:
+            # to ensure we don't get a KeyError
+            raise ValueError("Invalid quantization_config")
+        if (values["quantization_method"] and not v) or (not values["quantization_method"] and v):
+            raise ValueError("quantization_config and quantization_method must be provided together")
+        if not v:
+            return v
+
+        try:
+            return cls.dict_to_quantization_config(values["quantization_method"], v).to_dict()
+        except ImportError:
+            # we don't want to fail since the pass target might have the correct transformers version
+            logger.warning(
+                f"Could not import the config class for quantization method {values['quantization_method']}. Skipping "
+                " validation"
+            )
+            return v
+
+    def get_loading_args(self):
+        loading_args = {}
+        # copy args that can be directly copied
+        direct_copy_args = ["device_map", "max_memory"]
+        for arg in direct_copy_args:
+            if getattr(self, arg):
+                loading_args[arg] = deepcopy(getattr(self, arg))
+        # convert torch dtype to torch.dtype or "auto"
+        if self.torch_dtype:
+            loading_args["torch_dtype"] = self.get_torch_dtype()
+        # convert quantization_config to the config class
+        quantization_config = self.get_quantization_config()
+        if quantization_config:
+            loading_args["quantization_config"] = quantization_config
+        # add extra args
+        if self.extra_args:
+            loading_args.update(deepcopy(self.extra_args))
+        return loading_args
+
+    def get_torch_dtype(self):
+        v = self.torch_dtype
+        if isinstance(v, str) and v != "auto":
+            # get rid of torch. prefix, this might have been added when serializing
+            v = v.replace("torch.", "")
+            try:
+                return getattr(torch, v)
+            except AttributeError as e:
+                raise ValueError(f"Invalid torch dtype {v}") from e
+        return v
+
+    def get_quantization_config(self):
+        if not self.quantization_method or not self.quantization_config:
+            return None
+        return self.dict_to_quantization_config(self.quantization_method, self.quantization_config)
+
```

Comment:
could we directly using self.get_quantization_config since the validation will call dict_to_quantization_config?

Reply:
in this way, the method seems can be removed?

Reply:
validation actually calls `dict_to_quantization_config` but converts it back to dict at line 113. Otherwise, serialization and type checking becomes complicated. 

So I added this method to get the config object during model load time. Since `from_pretrained` doesn't actually work with dict for `gptq` quantization method.

Reply:
Resolving after offline conversion. tldr: there is no easy way to validate and serialize dataclasses (pydantic does things on top of them) + we don't have access to the parent class here for typing.